src/network/HeartbeatManager.ts

```diff
@@ -38,11 +39,13 @@ export class HeartbeatManager {
     private readonly heartbeatInterval: number;
     private logger: ILogger;
     private task: Task;
+    private readonly connectionRegistry: ConnectionRegistry;
```

Comment:
As we have the registry here, we can remove `connectionManager` field from `HeartbeatManager`. Hence, the circular dependency will be broken.